### PR TITLE
openscapes: Setup GitHub Auth for openscapes staging

### PR DESCRIPTION
**Pull request by yuvipanda**
- Switching to using orgs & teams for authentication
- Don't allow hand-entered usernames access anymore

Ref https://github.com/2i2c-org/infrastructure/issues/3240

### CODE REVIEW

**Comment by yuvipanda** [comment]
Will wait on @erinmr to provide list of orgs / teams to allow access

**Comment by erinmr** [comment]
GitHub teams are: 

- https://github.com/orgs/NASA-Openscapes/teams/workshopaccess-2i2c
- https://github.com/orgs/NASA-Openscapes/teams/longtermaccess-2i2c
- https://github.com/orgs/NASA-Openscapes/teams/championsaccess-2i2c

**Comment by yuvipanda** [comment]
ok, @erinmr has granted access and folks should be able to login to staging.openscapes.2i2c.cloud with these teams! We'll test then merge.

File: config/clusters/openscapes/common.values.yaml [hunk 1]
```diff
@@ -47,41 +47,17 @@ basehub:
...
-          #          access any more. In practice the user will still have
-          #          access if it had logged in once before, as it then exists
-          #          in JupyterHub's database of users.
-          #
-          allow_existing_users: True
+          authenticator_class: github
+        GitHubOAuthenticator:
+          oauth_callback_url: https://staging.openscapes.2i2c.cloud/hub/oauth_callback
+          allowed_organizations:
+            - 2i2c-org:hub-access-for-2i2c-staff
```
Comment (consideRatio) [inline config/clusters/openscapes/common.values.yaml:54]: With oauthenticator 16, we don't need this - its resolved by:

```yaml
        add_staff_user_ids_to_admin_users: true
        add_staff_user_ids_of_type: "github"
```

```suggestion
```

**Review by consideRatio — APPROVED** [review]

File: config/clusters/openscapes/common.values.yaml (hunk 1 above)
Comment (consideRatio) [reply config/clusters/openscapes/common.values.yaml:54]: Oh wait... I'll confirm my thoughts first. I made a search to confirm it by it was made incorrectly.

Comment (yuvipanda) [reply config/clusters/openscapes/common.values.yaml:54]: Thanks @consideRatio, I'll hold off for now.

Comment (yuvipanda) [reply config/clusters/openscapes/common.values.yaml:54]: Thanks @consideRatio. Just to verify, is that true when `OAuthenticator.allow_existing_users` is not set to True as well?

Comment (consideRatio) [reply config/clusters/openscapes/common.values.yaml:54]: https://github.com/2i2c-org/infrastructure/pull/3358 makes us 100% consistent

Comment (consideRatio) [reply config/clusters/openscapes/common.values.yaml:54]: > Thanks @consideRatio. Just to verify, is that true when `OAuthenticator.allow_existing_users` is not set to True as well?

Yes! With Oauthenticator 16 - a user allowed in any way is allowed, the union of ways to allow users as compared to the intersection.

`OAuthenticator.allow_existing_users` is one way to allow users, `admin_users`, `allowed_users`, `allowed_organizations` are other ways as well. This PR looks correct to me with the intent I've understood.

Comment (yuvipanda) [reply config/clusters/openscapes/common.values.yaml:54]: Thanks @consideRatio

**Pull request merged**